src/js/client/editprefset-viewport.js

```diff
@@ -0,0 +1,681 @@
+    that.events.openConfirmDeleteContextDialog.fire();
+};
+
+gpii.devpmt.editValueEvent = function (that, event) {
+    var newCurrent = {
+        context: event.currentTarget.dataset.context,
+        term: event.currentTarget.dataset.term,
+        //value: event.currentTarget.dataset.value,
+        product: event.currentTarget.dataset.product
+    };
+
+    var newMetadata = {};
+    if (!newCurrent.product) {
+        // Common Term
+        newMetadata = that.options.commonTerms[newCurrent.term];
+        newMetadata.name = newMetadata.schema.title;
+        newMetadata.description = newMetadata.schema.description;
+        newCurrent.value = that.lookupGenericPrefValue(newCurrent.context, newCurrent.term);
+        newCurrent.blank = newCurrent.value === undefined;
+    }
+    else {
+        // Application Specific
+        console.log(newCurrent);
+        // TODO ontology
+        newMetadata = gpii.devpmt.findProductSettingMetadata(that.model.allSolutions,
+            newCurrent.product.slice(38), newCurrent.term);
+        newMetadata.name = newMetadata.schema.title;
+        newMetadata.description = newMetadata.schema.description;
+        newCurrent.value = that.lookupProductPrefValue(newCurrent.context, newCurrent.product, newCurrent.term);
+        newCurrent.blank = newCurrent.value === undefined;
+    }
+    that.applier.change("currentlyEditing.current", newCurrent);
+    that.applier.change("currentlyEditing.metadata", newMetadata);
+    that.applier.change("currentlyEditing.active", true);
+    that.applier.change("unsavedChangesExist", true);
+    that.prefsAdjuster.renderInitialMarkup();
+    // $(document).foundation();
+    // https://github.com/zurb/foundation-sites/issues/7899
+    $(".sticky").foundation("_calc", true);
+};
+
+
+/**
+ * Temporary workaround until we set up an ajax feed for these. The server
+ * side template is rendering the transformed preferences in another script
+ * element at the top of the page.
+ */
+gpii.devpmt.npsetInit = function (that) {
+    // Should use change applier
+    that.model.flatPrefs = that.options.npset.options.flatPrefs;
+    that.model.commonTerms = that.options.commonTerms;
+    that.model.contextNames = gpii.devpmt.contextNames(that.options.npset.options.flatPrefs);
+    that.model.npsetApplications = gpii.devpmt.npsetApplications(that.options.npset.options.flatPrefs);
+    that.model.docs = that.options.npset.options.docs;
+    that.model.npsetName = that.options.npset.options.npsetName;
+    that.model.allSolutions = that.options.allSolutions;
+
+    // Add sorted solutions
+    var allSolutionsSorted = [];
+    // that.model.allSolutionsSorted = [];
+    fluid.each(that.model.allSolutions, function (item, key) {
+        var flatItem = fluid.copy(item);
+        flatItem.id = key;
+        allSolutionsSorted.push(flatItem);
+    });
+
+    var sortName = function (a, b) {
+        if (a.name > b.name) {
```

Comment:
Less clunky as well as supporting Unicode: https://stackoverflow.com/a/8900922

Reply:
this is using localeCompare